python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1762,6 +1763,105 @@ def getURI(self, ref: DatasetRef, predict: bool = False) -> ResourcePath:
             )
         return primary
 
+    def getManyURIs(
+        self,
+        refs: Iterable[DatasetRef],
+        predict: bool = False,
+        throwIfNotFound: bool = True,
+    ) -> Dict[DatasetRef, DatasetRefURIs]:
+        # Docstring inherited
+
+        PrimaryType = Optional[ResourcePath]
+        ComponentsType = Dict[str, ResourcePath]
+        uris: Dict[DatasetRef, DatasetRefURIs] = {}
+
+        records = self._get_stored_records_associated_with_refs(refs)
+
+        missingRefs = [ref for ref in refs if ref.id not in records.keys()]
+
+        for ref in missingRefs:
+            # if this has never been written then we have to guess
```

Comment:
This code has all been copied from `getURIs()`. We should not be doing that. Better to call `self.getURIs()` directly. The one reason not to do that is that we already know that `self.exists(ref)` is going to fail and so it would be more efficient to not call that at all.

The quick fix is to delete this code and live with doing an extra call to exists() that we know is going to fail.

The efficient thing to do is to factor out the URI prediction code so that getURIs can call it and getManyURIs can call it with getManyURIs calling it without having to do the self.exists check.